### PR TITLE
[Cisco ASA] Support 113015 event and hidden username value

### PR DESCRIPTION
**Pull request by chemamartinez**
## Proposed commit message

Added two changes for the Cisco ASA integration:
- Support for 113015 event message. More information about the event format [here](https://www.cisco.com/c/en/us/td/docs/security/asa/syslog/b_syslog/syslog-messages-101001-to-199021.html#con_4769508).
- Accept `*****` as a valid Cisco user name. According to the [Cisco documentation](https://www.cisco.com/c/en/us/td/docs/security/asa/syslog/b_syslog/syslog-messages-101001-to-199021.html#con_4769508):
```
The username is hidden when invalid or unknown, but appears when valid or the no logging hide username command has been configured.
```
It means some events could contain a hidden username represented by `*****`. These events generated a pipeline error since that was not a valid username in the grok pattern.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Added logs for the pipeline tests to cover the new changes.

### CODE REVIEW

File: packages/cisco_asa/data_stream/log/_dev/test/pipeline/test-additional-messages.log-expected.json [hunk 1]
```diff
@@ -7948,6 +7948,164 @@
...
+            "related": {
+                "hosts": [
+                    "fw123-vc456"
+                ],
+                "ip": [
+                    "81.2.69.143"
+                ],
+                "user": [
+                    "*****"
+                ]
```
Comment (efd6) [inline packages/cisco_asa/data_stream/log/_dev/test/pipeline/test-additional-messages.log-expected.json:8085]: Should we remove this?

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -456,15 +456,24 @@ processors:
...
+        CISCO_USER: (\*\*\*\*\*|((LOCAL\\)?(%{HOSTNAME}\\)?%{USERNAME}(@%{HOSTNAME})?(, *%{NUMBER})?))
+  - grok:
+      if: "ctx._temp_.cisco.message_id == '113015'"
+      field: "message"
+      description: "113015"
+      patterns:
+      - "AAA user authentication Rejected(%{SPACE})?: reason = %{REASON:_temp_.cisco.rejection_reason}(%{SPACE})?: local database(%{SPACE})?: [Uu]ser = %{CISCO_USER:source.user.name}(%{SPACE})?: [Uu]ser IP = %{IP:source.address}"
+      pattern_definitions:
+        REASON: (AAA failure|Account has been disabled|Invalid password|Password is expiring|Password has expired|Password malformed|Unspecified)
+        CISCO_USER: (\*\*\*\*\*|(\*\*\*\*\*|((LOCAL\\)?(%{HOSTNAME}\\)?%{USERNAME}(@%{HOSTNAME})?(, *%{NUMBER})?)))
```
Comment (efd6) [inline packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml:476]: Are both of these alternatives required?

File: packages/cisco_asa/data_stream/log/_dev/test/pipeline/test-additional-messages.log-expected.json (hunk 1 above)
Comment (chemamartinez) [reply packages/cisco_asa/data_stream/log/_dev/test/pipeline/test-additional-messages.log-expected.json]: Sure, good point!

File: packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (chemamartinez) [reply packages/cisco_asa/data_stream/log/elasticsearch/ingest_pipeline/default.yml]: No, it has been a mistake when replicating the change.

**Review by efd6 — APPROVED** [review]

**Pull request merged**